rustyms-py/src/lib.rs

```diff
@@ -0,0 +1,1262 @@
+    ///
+    /// Parameters
+    /// ----------
+    /// max_charge : int
+    ///    The maximal charge of the fragments.
+    /// model : str
+    ///   The model to use for the fragmentation. One of: ``all``, ``cid_hcd``, ``etcid``, ``etd``, ``ethcd``
+    ///
+    /// Returns
+    /// -------
+    /// list[Fragment]
+    ///   The theoretical fragments.
+    ///
+    fn generate_theoretical_fragments(
+        &self,
+        max_charge: i16,
+        model: &str,
+        // peptide_index: usize, TODO: Required for linear peptide?
+    ) -> PyResult<Vec<Fragment>> {
+        let model = match_model(model)?;
+        Ok(self
+            .0
+            .generate_theoretical_fragments(
+                rustyms::system::Charge::new::<rustyms::system::e>(max_charge as f64),
+                &model,
+                0, // TODO: Don't hard code?
+            )
+            .unwrap()
+            .iter()
+            .map(|f| Fragment(f.clone()))
+            .collect())
+    }
+}
+
+#[pyclass]
+struct RawPeak(rustyms::spectrum::RawPeak);
+
+#[pymethods]
+impl RawPeak {
+    fn __repr__(&self) -> String {
+        format!(
+            "RawPeak(charge={}, mz={}, intensity={})",
+            self.charge(),
+            self.mz(),
+            self.intensity()
+        )
+    }
+
+    /// The charge of the peak.
+    ///
+    /// Returns
+    /// -------
+    /// int
+    ///
+    #[getter]
+    fn charge(&self) -> i16 {
+        self.0.charge.value as i16
+    }
+
+    /// The m/z value of the peak.
+    ///
+    /// Returns
+    /// -------
+    /// float
+    ///
+    #[getter]
+    fn mz(&self) -> f64 {
+        self.0.mz.value
+    }
+
+    /// The intensity of the peak.
+    ///
+    /// Returns
+    /// -------
+    /// float
+    ///
+    #[getter]
+    fn intensity(&self) -> f64 {
+        self.0.intensity
+    }
+}
+
+#[pyclass]
+/// Represents an annotated peak in a mass spectrometry spectrum.
+struct AnnotatedPeak(rustyms::spectrum::AnnotatedPeak);
+
+#[pymethods]
+impl AnnotatedPeak {
+    fn __repr__(&self) -> String {
+        format!(
+            "AnnotatedPeak(experimental_mz={}, intensity={}, charge={}, annotation=[{:?}])",
+            self.experimental_mz(),
+            self.intensity(),
+            self.charge(),
+            self.annotation(),
+        )
+    }
+
+    /// The experimental m/z value of the peak.
+    ///
+    /// Returns
+    /// -------
+    /// float
+    ///
+    #[getter]
+    fn experimental_mz(&self) -> f64 {
+        self.0.experimental_mz.value
+    }
+
+    /// The intensity of the peak.
+    ///
+    /// Returns
+    /// -------
+    /// float
+    ///
+    #[getter]
+    fn intensity(&self) -> f64 {
+        self.0.intensity
+    }
+
+    /// The charge of the peak.
+    ///
+    /// Returns
+    /// -------
+    /// int
+    ///
+    #[getter]
+    fn charge(&self) -> i16 {
+        self.0.charge.value as i16
+    }
+
+    /// The annotation of the peak, if present.
```

Comment:
This is a list, so I would say: "all annotations of this peaks (there can be none)".